CodeOfConduct.md

```diff
@@ -16,11 +16,46 @@ WebAssembly operates under the W3C's
 > vehicle for better identity of the organization.
 
 We hope that our community group act according to these guidelines, and that
-participants hold each other to these high standards. If you have any questions
-or are worried that the code isn't being followed, please contact the Community
-Group's chairs at `team-wasm-chairs@w3.org` (note: this list is also
-visible to W3C staff). For very serious concerns, the W3C has [procedures][]
-allowing you to access its ombuds directly and confidentially.
+participants hold each other to these high standards. 
+
+If you are worried that the code isn't being followed, or are witness to a violation but are
+uncomfortable speaking up,please contact the Community Group's chairs at
+`team-wasm-chairs@w3.org` (note: this list is also visible to W3C staff). 
+
+In cases where there is repeated disregard for the CEPC, it is possible to incur disciplinary
+action. The points below outline a series of steps that will be taken in case of violations
+to ensure a positive work environment in the WebAssembly CG. 
+
+ - **Informal warning**   
+   An informal or verbal warning can be issued in any relevant medium of communication callin 
+   out inappropriate behavior as a violation of the CEPC
+   (examples: responses to github issues or PRs, discord, email, meetings).
+
+ - **Corrective action and/or counseling**   
+   Informal meetings, or email discussions where problematic behavior is addressed,
+   and suggestions for constructive ways forward are offered. A mediation process can also be an
+   option if relevant to the situation and both parties agree. 
+
+ - **Formal warning**   
+   An explicit warning issued after repeated incidents. While members of the CG can call out
+   inappropriate behavior as needed, warnings will be issued by CG chairs, subgroup chairs, or
+   proposal champions depending on the nature of the violation.
```

Comment:
What kind of reporting is expected from the chairs or champions giving out formal warnings? Can chairs and champions unilaterally give out formal warnings or is there some group that needs to be consulted first?

Reply:
This document is intentionally setup a a set of escalations, with persistent violations of the CEPC, I would expect that any formal warning if not publicly issued, will go through the chairs. I'll reword this point to be more clear.